### ISSUE TITLE
Development/maintenance status

### ISSUE THREAD

**Issue by sotte** [ActiveState/appdirs#79]
What is the development status of this project? The last commit was on Jul 23, 2015, but there seem to be some interest in this project (issues as well as PRs).

Maybe it's possible to find a new maintainer for this project you guys don't want to keep maintaining it.

**Comment by Kriskras99** [comment]
I would love to take over this project.
In my fork I have implemented new tests and merged the current PRs. And in the dev branch I have added support for xdg-user-dirs and replaced CSDIL with KNOWNFOLDERID.

**Comment by zoofood** [comment]
Hi Guys,

I've discussed with the team here at ActiveState and we'll ramp up maintenance again on this shortly. Sorry for the delay.

Cheers,

-JR

**Comment by sotte** [comment]
Nice to hear that!

Maybe you can integrate @Kriskras99 changes.

**Comment by Kriskras99** [comment]
I will open a pull request when the other pull reqiests are merged or closed, because I depend on some of them.

**Comment by Kriskras99** [comment]
@zoofood could you start merging the already open pull requests (Especially #53, #56 and #64)? Then I could open a pull request for a new test framework, and give a more "correct" replacement for CSIDL.

**Comment by dgentry** [comment]
Hi @zoofood, 

Do you know when "shortly" is?  If it's going to be more than another week (just over a month since you last posted), maybe you could just add @Kriskras99's as a repo maintainer?  If not, maybe we should all consider his stable fork (https://github.com/Kriskras99/appdirs/tree/stable) to be the canonical one for a while.  At my company, twoporeguys.com, we would immediately use #64, and #81.

Thanks,
Dennis

**Comment by zoofood** [comment]
My apologies. The team was very busy, I'm moving things through here myself, kicking it off over the holidays here, but will take a bit of ramp up.

**Comment by Kriskras99** [comment]
Nice to hear, as soon as the pull requests are merged, I will squash my commits and open a pull request.

**Comment by hobarrera** [comment]
TBH, since development has stagnated before (and picking this up again is also going a bit slow), maybe you ought to consider taking up a co-maintainer, so we're sure this doesn't happen again (it also increases trust by users/other devs).

**Comment by pradyunsg** [comment]
@zoofood Please do consider adding someone as a co-maintainer -- it would help reduce the bus factor here and improve the health of the project as a whole.

**Comment by pradyunsg** [comment]
@zoofood I'll be happy to be a co-maintainer for this project, if you're willing to add one. :)

**Comment by jhermann** [comment]
https://jazzband.co/ would be another option. 27 commits since the last release that could be published…

**Comment by ofek** [comment]
I agree that https://github.com/jazzband seems like the logical home

**Comment by pradyunsg** [comment]
Dropping in to say that I'm still interested in picking up maintenance for this package.

**Comment by McSinyx** [comment]
@pradyunsg: I think it's not that the @ActiveState team cannot maintain this, but they are like completely off the grid (sure @zoofood does not seem to visit GitHub very regularly, but there should be at least another team member with write access, or even just some minimal communication).  It was really sad to see @ofek pinging for almost two weeks straight.  I'll not be surprised to see you dropping the same message next year and still be entirely ignored.

**Comment by zoofood** [comment]
Hi Gang, 

I appreciate everyone's anxiousness/frustration/concern here. I originally took on maintenance as Trent and Srid had moved on and it needed a home and we were rebuilding our team at ActiveState. It has sadly languished at the bottom of my todo list in the last while. It is not any issue with ActiveState per se, we have plenty of community initiatives spread across many language ecosystems (Python/Perl/Go/Tcl) that we are putting time/money into and not the least of which we are building a platform by which the community can put together their own Python runtimes and dependencies, all built from source.

I always planned to get back to this project as I quite enjoy it and want to give back where possible. With that in mind, I am going to work internally at ActiveState to find some resources to get this project back in shape. If we can't put a meaningful effort into this project and get releases out we will hand it off by the end of April.

I have had several offers of help and many good options, thanks to everyone that has reached out.

Thanks.

-JR

**Comment by ofek** [comment]
@zoofood Any update?

**Comment by zoofood** [comment]
Hi @ofek, thanks for checking in. Had some hurdles to get it on PyPI, will be resolved this week and the first of several updates will go out to get us current.

**Comment by pradyunsg** [comment]
*goes to https://pypi.org/project/appdirs/*

> appdirs 1.4.4
>
> Released: May 11, 2020

Hurray! 🎉

**Comment by ofek** [comment]
@zoofood Are we still considering donating the project?

I see a release happened, which is great! However, the only other development has been merging a shebang line removal and 2 PRs removing names of people who no longer maintain this...

**Comment by pradyunsg** [comment]
My offer of picking up the maintenance of this package still stands and I'm still interested.

I'll also echo the suggestion of making this a part of jazzband to ease the development overhead.

**Comment by eukreign** [comment]
I just joined jazzband (in anticipation of this project being moved over there) and thus also support moving this project there.

I have a PR pending on this repo that's been there for a week (and I see dozens others for way longer): https://github.com/ActiveState/appdirs/pull/147

I work on a product which has been using `appdirs` (https://github.com/lbryio/lbry-sdk) for at least 4 years, so am very invested in continuing to maintain this project, ideally jazzband can adopt it, otherwise I'll have to fork it and contribute to the existing 70 forks (i'd rather not do that).

I can take the lead on this or support someone else here who's been pushing for this before me.

Let's just do something.

**Comment by ofek** [comment]
Any update?

**Comment by ofek** [comment]
I guess not?

**Comment by gaborbernat** [comment]
> I always planned to get back to this project as I quite enjoy it and want to give back where possible. With that in mind, I am going to work internally at ActiveState to find some resources to get this project back in shape. If we can't put a meaningful effort into this project and get releases out we will hand it off by the end of April.

I understand that resource availability is an issue. However, considering this I think everyone would benefit if you'd bless 2 more people to maintain besides yourself. 17 pull requests open does not feel like a good place to be. Thanks for your efforts until now but consider offloading some of the work when you get busy with other things.

**Comment by ofek** [comment]
@zoofood Any progress on this?

**Comment by pradyunsg** [comment]
Yet-another-nudge.

**Comment by ofek** [comment]
Any update? @zoofood

**Comment by pradyunsg** [comment]
_nudge_

**Comment by pradyunsg** [comment]
> I'm up for picking up the maintenance of this package (my earlier offer to co-maintain still stands) and I'm still interested. I'll also echo the suggestion of making this a part of jazzband to ease the development overhead.

Reiterating that this offer still stands.

**Comment by ofek** [comment]
Any update? @zoofood

**Comment by gaborbernat** [comment]
At this point, I'm tempted to fork the project and start maintaining it under a new name 🤔

**Comment by WhyNotHugo** [comment]
@gaborbernat Looks like there's been a bit of activity in recent weeks here. Might be worth seeing how this moves forward.

**Comment by ofek** [comment]
@gaborbernat Can you please? We'll switch immediately.

**Comment by ofek** [comment]
@gaborbernat Did you fork?

**Comment by gaborbernat** [comment]
@ofek not yet 😊 if you'd want this now you could go-ahead 👍🏻

**Comment by YariKartoshe4ka** [comment]
I am interesing in this project too and do not mind be maintencer or help it with PRs

**Comment by gaborbernat** [comment]
For anyone looking we forked the repository to https://github.com/platformdirs/platformdirs 👍

**Comment by WhyNotHugo** [comment]
Whoops. I've been working on `apppaths` as a fork. Maybe we should coordinate better, do you think an IRC channel to follow up on direction an alike would make sense?

I'd rather hold on my fork if there's already another -- having two does more damage than harm. I don't honestly have many changes, mostly updating the code to return `Path` instances rather than `str` instances.

**Comment by gaborbernat** [comment]
Feel free to open discussions at https://github.com/platformdirs/platformdirs

**Comment by WhyNotHugo** [comment]
Looks like issues for that repository are disabled.

**Comment by ofek** [comment]
I've just enabled them, try now.

**Comment by elenakrittik** [comment]
☠️

**Comment by pradyunsg** [comment]
On a different note, we can close this out on the basis that https://github.com/ActiveState/appdirs/commit/8734277956c1df3b85385e6b308e954910533884 has happened.
